src/PIL/Image.py

```diff
@@ -1949,6 +1955,57 @@ def save(self, fp, format=None, **params):
             if open_fp:
                 fp.close()
 
+    def _convert_mode(self, plugin, params):
+        if not hasattr(plugin, '_supported_modes'):
+            return
+        supported_modes = plugin._supported_modes()
+        if not supported_modes or self.mode in supported_modes:
+            return
+        if self.mode == 'P':
+            preferred_modes = []
+            if 'A' in self.im.getpalettemode():
+                preferred_modes.append('RGBA')
+            preferred_modes.append('RGB')
+        else:
+            preferred_modes = {
+                'RGBA': ['RGB'],
+                'LA': ['RGBA', 'P', 'L'],
+                'L': ['RGB'],
+                'I': ['L', 'RGB'],
+                'CMYK': ['RGB']
```

Comment:
There should be a full list of possible modes, for example 'YCbCr' and others.

Reply:
Okay, I've updated the list, see what you think